src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,200 @@
+  %               options can be shown in litteral like ``this`` or ``that``.
+  % :type argin2: string
+  % :param argin3: (dimension) optional argument
+  %
+  % :returns: - :argout1: (type) (dimension)
+  %           - :argout2: (type) (dimension)
+  %
+  % .. todo:
+  %
+  %    - expand to run through multiple subjs ans groups
+  %      (https://stackoverflow.com/questions/8748976/list-the-subfolders-in-a-folder-matlab-only-subfolders-not-files)
+  %    - generalize how to retrieve RT from sidecar json file
+  %    - saveMatlabBatch(matlabbatch, ['3Dto4D_dataType-' num2str(dataType) '_RT-' num2str(RT)], opt, subID);
+  
+  % Add folder where some functions live
+  initEnv()
+
+  % Set the folder where sequences folders exist
+  optSource.sourceDir = '/Users/barilari/Desktop/DICOM_UCL_leuven/renamed/sub-pilot001/ses-002/MRI';
+
+  % List of the sequences that you want to skip (folder name pattern)
+  optSource.sequenceToIgnore = {'AAHead_Scout', ...
+                                'b1map', ...
+                                't1', ...
+                                'gre_field'};
+
+  % Number of volumes to discard ad dummies, (0 is default)
+  optSource.nbDummies = 5;
+
+  % List of the sequences where you want to remove dummies (folder name pattern)
+  optSource.sequenceRmDummies = {'cmrr_mbep2d_p3_mb2_1.6iso_AABrain', ...
+                                 'cmrr_mbep2d_p4_mb2_750um_AAbrain'};
+
+  % Set data format conversion (0 is default)
+
+  % 0:  SAME
+  % 2:  UINT8   - unsigned char
+  % 4:  INT16   - signed short
+  % 8:  INT32   - signed int
+  % 16: FLOAT32 - single prec. float
+  % 64: FLOAT64 - double prec. float
+
+  optSource.dataType = 0;
+
+  % Boolean to enable gzip of the new 4D file (0 is default)
+  optSource.zip = 0;
+
+  % Check the options provided
+  optSource = checkOptionsSource(optSource);
+
+  % Get source folder content
+  sourceDataStruc = dir(optSource.sourceDir);
+
+  isDir = [sourceDataStruc(:).isdir];
+
+  optSource.sequenceList = {sourceDataStruc(isDir).name}';
+
+  % Loop through the sequence folders
+
+  tic;
+
+  for iSeq = 1:size(optSource.sequenceList, 1)
+
+    % Skip 'non' folders
+    if length(char(optSource.sequenceList(iSeq))) > 2
+
+      % Check if sequence to ignore or not
+      if contains(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)
+
+        newline;
+        warning('IGNORING SEQUENCE: %s', string(optSource.sequenceList(iSeq)));
+        newline;
+
+      else
+
+        fprintf('\n\nCONVERTING SEQUENCE: %s \n\n', char(optSource.sequenceList(iSeq)));
+
+        % Set whether to remove dummies or not
+
+        nbDummies = 0;
+
+        if contains(optSource.sequenceList(iSeq), optSource.sequenceRmDummies)
+
+          nbDummies = optSource.nbDummies;
+
+          fprintf('\n\nREMOVING %s DUMMIES\n\n', num2str(nbDummies));
+
+        end
+
+        % Get sequence folder path
+        sequencePath = char(fullfile(optSource.sourceDir, optSource.sequenceList(iSeq)));
```

Comment:
do you need a `char` here ?

I think that `fullfile` should return a string anyway, no?

Reply:
hum...

I think that the issue is that `optSource.sequenceList` is a cell.

So if you call `optSource.sequenceList(iSeq)` it will return a cell.

But if you call optSource.sequenceList **{** iSeq **}** it will return a string.

I think that's why you need to use `char` in many places in the code.

For example a bit lower:

```

          delete(char(volumesList(iDel)));
          delete(char(jsonList(iDel)));
```

Could that be the reason?

Reply:
good catch :)